app.ts

```diff
@@ -22,11 +22,15 @@ class AjaxCall {
 	}
 
 	onRStateChange(callback: Function) {
-		if (this.objHttpReq.readyState === 4 && this.objHttpReq.status === 200) {
-			let data = JSON.parse(this.objHttpReq.responseText);
-			(callback)(data, this.indexStart, this.indexFinish);
+		if (this.objHttpReq.readyState === XMLHttpRequest.DONE) {
+			if (this.objHttpReq.status === 200) {
+				let data = JSON.parse(this.objHttpReq.responseText);
+				(callback)(data, this.indexStart, this.indexFinish);
+			}
+			else {
+				console.log("AJAX error with the following status occurred: " + this.objHttpReq.status);
```

Comment:
Wow harsh... ;) Jeremy is correct that we use [little wrappers functions](https://github.com/IMQS/www/blob/master/js/js-base/defs.ts#L113-L178) for `console.log` inside the www codebase: `iq.logErr` and its friends `iq.logWarn`, `iq.logInfo`, `iq.logDebug` and `iq.logTrace`. For the onboarding project, however, you don't have access to those functions, so using `console.log` in this case is completely fine. 

The reason we don't use it directly in our production codebase is because if the IT department at a client deactivated the developer tools in IE, `console.log` would cause an error. 

On the amount of info you log: I would say feel free to spam away, more info is better than none, just make sure to use the correct log level.